File: jdevkit/src/main/java/com/github/klee0kai/android_devkit/collections/weaklist/WeakList.java
package com.github.klee0kai.android_devkit.collections.weaklist;

import com.github.klee0kai.android_devkit.model.IProvide;
import com.github.klee0kai.android_devkit.model.RefProvide;

import java.util.Collection;

public class WeakList<T> extends RefList<T> {

    public WeakList() {

    }

    public WeakList(Collection<? extends T> c) {
        this();
        addAll(c);
    }


    @Override
    IProvide<T> wrapRef(T val) {
        return RefProvide.weak(val);
    }
}

File: kdevkit/src/main/kotlin/com/github/klee0kai/android_devkit/collections/weaklist/WeakList.kt
package com.github.klee0kai.android_devkit.collections.weaklist

import com.github.klee0kai.android_devkit.model.IProvide
import com.github.klee0kai.android_devkit.model.RefProvide

class WeakList<T> constructor() : RefList<T>() {

    constructor(list: Iterable<T?>) : this() {
        addAll(list)
    }

    override fun wrapRef(it: T?): IProvide<T?> {
        return RefProvide.weak(it)
    }

    override fun createNew(list: List<T?>): RefList<T> {
        return WeakList(list)
    }
}